packages/angular/cli/src/commands/update/cli.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Argv } from 'yargs';
+import {
+  CommandModule,
+  CommandScope,
+  Options,
+  OtherOptions,
+} from '../../command-builder/command-module';
+import { UpdateCommand } from './update-impl';
+
+export interface UpdateCommandArgs {
+  packages?: string | string[];
+  force: boolean;
+  next: boolean;
+  'migrate-only'?: boolean;
+  name?: string;
+  from?: string;
+  to?: string;
+  'allow-dirty': boolean;
+  verbose: boolean;
+  'create-commits': boolean;
+}
+
+export class UpdateCommandModule extends CommandModule<UpdateCommandArgs> {
+  static override scope = CommandScope.In;
+
+  command = 'update [packages..]';
+  describe = 'Updates your workspace and its dependencies. See https://update.angular.io/.';
+  longDescriptionPath?: string | undefined;
+
+  builder(localYargs: Argv): Argv<UpdateCommandArgs> {
+    return localYargs
+      .positional('packages', {
+        description: 'The names of package(s) to update.',
+        type: 'string',
+      })
+      .option('force', {
+        description:
+          'Ignore peer dependency version mismatches. ' +
+          'Passes the `--force` flag to the package manager when installing packages.',
+        type: 'boolean',
+        default: false,
+      })
+      .option('next', {
+        description: 'Use the prerelease version, including beta and RCs.',
+        type: 'boolean',
+        default: false,
+      })
+      .option('migrate-only', {
+        description: 'Only perform a migration, do not update the installed version.',
+        type: 'boolean',
+      })
+      .option('name', {
+        description: 'The name of the migration to run.',
+        type: 'string',
+        implies: ['migrate-only'],
+        conflicts: ['to', 'from'],
+      })
```

Comment:
Could the `coerce` yargs functionality on `migrate-only` be used to allow for a deprecation time period of the `--migrate-only <string>` scenario?

Reply:
Unfortunately not, since the type of the option is boolean, the value will be covered to a boolean by the parser which happens way before `coerce` being triggered.